src/bacnet/basic/object/bi.c

```diff
@@ -908,7 +908,9 @@ int Binary_Input_Read_Property(BACNET_READ_PROPERTY_DATA *rpdata)
         case PROP_STATUS_FLAGS:
             /* note: see the details in the standard on how to use these */
             bitstring_init(&bit_string);
-            bitstring_set_bit(&bit_string, STATUS_FLAG_IN_ALARM, false);
+            bitstring_set_bit(
+                &bit_string, STATUS_FLAG_IN_ALARM,
+                pObject->Event_State != EVENT_STATE_NORMAL);
```

Comment:
please use  `Binary_Input_Event_State(rpdata->object_instance)` since pObject isn't guaranteed to be available (it is behind some `#ifdef`).

Reply:
Will do... May I ask why? the direct access through pointer is way more efficien...